_episodes/04-use-license.md

```diff
@@ -5,7 +5,185 @@ exercises: 0
+
+Most of the open source licenses can be divided into two big groups: copyleft (or share-alike) and permissive. Copyleft licenses includes a clause that guarantees continued open access to a software and its source code in derivative works. The [GNU General Public License](https://www.gnu.org/licenses/gpl-3.0.en.html) is the most famous copyleft license and it says
+
+> To protect your rights, we need to prevent others from denying you these rights or asking you to surrender the rights. Therefore, you have certain responsibilities if you distribute copies of the software, or if you modify it: responsibilities to respect the freedom of others.
+>
+> For example, if you distribute copies of such a program, whether gratis or for a fee, you must pass on to the recipients the same freedoms that you received. You must make sure that they, too, receive or can get the source code. And you must show them these terms so they know their rights.
+
+Permissive licenses, as opposed to copyleft licenses, does not enforce that derivative works are licensed under the same terms or compatible terms of the original work. Famous examples of permissive licenses are the [MIT](https://opensource.org/licenses/MIT) and [BSD](https://opensource.org/licenses/BSD).
+
+![Diagram](http://journals.plos.org/ploscompbiol/article/figure/image?size=large&download=&id=10.1371/journal.pcbi.1002598.g002)
+
+> ### Challenge 1
+>
+> Work in pairs. Pick one item from the list below. Discuss with your partner what would be the implications to your project if one of the third-party libraries that you are considering to use, has the fragment mentioned in the selected item as part of their license.
+> 
+> 1. "Linking this library statically or dynamically with other modules is making a combined work based on this library. Thus, the terms and conditions of the GNU General Public License cover the whole combination."
+>
+>    Source: GNU Classpath's License
+> 2. "The license granted hereunder will terminate, automatically and without notice, if you (or any of your subsidiaries, corporate affiliates or agents) initiate directly or indirectly, or take a direct financial interest in, any Patent Assertion: (...)"
+>
+>    Source: [React's License at 10 April 2015](https://github.com/facebook/react/blob/b8ba8c83f318b84e42933f6928f231dc0918f864/PATENTS).
+> 3. "(...) each Contributor hereby grants to You a perpetual, worldwide, non-exclusive, no-charge, royalty-free, irrevocable (except as stated in this section) patent license to make, have made, use, offer to sell, sell, import, and otherwise transfer the Work, where such license applies only to those patent claims licensable by such Contributor that are necessarily infringed by their Contribution(s) alone or by combination of their Contribution(s) with the Work to which such Contribution(s) was submitted."
+>
+>    Source: [Apace 2.0 License](https://opensource.org/licenses/Apache-2.0)
+>
```

Comment:
I asked one of my colleagues (who know a lot more about licenses than I do) to have a quick look at the challenges on this episode. He suggested adding these two questions:

 - "Who owns the copyright to the various programs you've written so far?"
 - "What are the (dis)advantages to licensing your software under the Apache license versus licensing it under the GPL"?

I am guessing this might be a good place to add these questions ?